runtime/common/src/claims.rs

```diff
@@ -336,6 +464,77 @@ impl<T: Trait> sp_runtime::traits::ValidateUnsigned for Module<T> {
 	}
 }
 
+/// Validate `attest` calls prior to execution. Needed to avoid a DoS attack since they are
+/// otherwise free to place on chain.
+#[derive(Encode, Decode, Clone, Eq, PartialEq)]
+pub struct PrevalidateAttests<T: Trait + Send + Sync>(sp_std::marker::PhantomData<T>) where
+	<T as system::Trait>::Call: IsSubType<Module<T>, T>;
+
+impl<T: Trait + Send + Sync> Debug for PrevalidateAttests<T> where
+	<T as system::Trait>::Call: IsSubType<Module<T>, T>
+{
+	#[cfg(feature = "std")]
+	fn fmt(&self, f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		write!(f, "PrevalidateAttests")
+	}
+
+	#[cfg(not(feature = "std"))]
+	fn fmt(&self, _: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		Ok(())
+	}
+}
+
+impl<T: Trait + Send + Sync> PrevalidateAttests<T> where
+	<T as system::Trait>::Call: IsSubType<Module<T>, T>
+{
+	/// Create new `SignedExtension` to check runtime version.
+	pub fn new() -> Self {
+		Self(sp_std::marker::PhantomData)
+	}
+}
+
+impl<T: Trait + Send + Sync> SignedExtension for PrevalidateAttests<T> where
+	<T as system::Trait>::Call: IsSubType<Module<T>, T>
+{
+	type AccountId = T::AccountId;
+	type Call = <T as system::Trait>::Call;
+	type AdditionalSigned = ();
+	type Pre = ();
+	const IDENTIFIER: &'static str = "PrevalidateAttests";
+
+	fn additional_signed(&self) -> Result<Self::AdditionalSigned, TransactionValidityError> {
+		Ok(())
+	}
+
+	fn validate(
+		&self,
+		who: &Self::AccountId,
+		call: &Self::Call,
+		_info: &DispatchInfoOf<Self::Call>,
+		_len: usize,
+	) -> TransactionValidity {
+		if let Some(local_call) = call.is_sub_type() {
+			if let Call::attest(attested_statement) = local_call {
+				let signer = match Preclaims::<T>::get(who) {
+					Some(x) => x,
+					None =>
+						return Err(InvalidTransaction::Custom(
+							ValidityError::SignerHasNoClaim.into()
+						).into()),
+				};
```

Comment:
```suggestion
				let signer = match Preclaims::<T>::get(who).ok_or(
					InvalidTransaction::Custom(ValidityError::SignerHasNoClaim.into()
				)?;
```
would something like this work? seems simpler